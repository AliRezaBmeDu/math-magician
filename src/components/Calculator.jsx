import React, { useState } from 'react';
import Display from './Display';
import calculate from '../logic/calculate';

const Calculator = () => {
  const initialDisplay = {
    total: '',
    next: null,
    operation: null,
  };

  const [result, setResult] = useState(initialDisplay);

  const handleClick = (e) => {
    const buttonName = e.target.textContent;
    if (!result && buttonName.match(/[0-9]+/)) {
      setResult({
        total: buttonName,
        next: null,
        operation: null,
      });
    } else {
      localStorage.setItem('current', '');
    }

    setResult(calculate(result, buttonName));
  };

  return (
    <div className="grid-container">
      <Display data={result.total} />
      <div className="row second">
        <button type="button" onClick={handleClick}>AC</button>
        <button type="button" onClick={handleClick}>+/-</button>
        <button type="button" onClick={handleClick}>%</button>
        <button type="button" onClick={handleClick}>รท</button>
      </div>
      <div className="row third">
        <button type="button" onClick={handleClick}>7</button>
        <button type="button" onClick={handleClick}>8</button>
        <button type="button" onClick={handleClick}>9</button>
        <button type="button" onClick={handleClick}>x</button>
      </div>
      <div className="row fourth">
        <button type="button" onClick={handleClick}>4</button>
        <button type="button" onClick={handleClick}>5</button>
        <button type="button" onClick={handleClick}>6</button>
        <button type="button" onClick={handleClick}>-</button>
      </div>
      <div className="row fifth">
        <button type="button" onClick={handleClick}>1</button>
        <button type="button" onClick={handleClick}>2</button>
        <button type="button" onClick={handleClick}>3</button>
        <button type="button" onClick={handleClick}>+</button>
      </div>
      <div className="row sixth">
        <button type="button" onClick={handleClick}>0</button>
        <button type="button" onClick={handleClick}>.</button>
        <button type="button" onClick={handleClick}>=</button>
      </div>
    </div>
  );
};

export default Calculator;
