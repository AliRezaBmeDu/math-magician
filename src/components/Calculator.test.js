import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from './Calculator';

it('should render without crashing', () => {
  const component = renderer.create(
    <Calculator />,
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});