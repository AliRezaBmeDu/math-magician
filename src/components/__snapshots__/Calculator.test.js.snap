// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<div
  className="grid-container"
>
  <div
    className="row first"
  >
    <span
      className="full-span"
    >
      0
    </span>
  </div>
  <div
    className="row second"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="row third"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>
  <div
    className="row fourth"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="row fifth"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="row sixth"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
